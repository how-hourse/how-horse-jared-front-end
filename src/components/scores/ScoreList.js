import React from 'react';
import PropTypes from 'prop-types';
import Score from './Score';

function ScoreList({ scores }) {
  const liStyle = {
    padding: '0px'
  };
  const scoreArr = scores.map(score => {
    return (
      <li style={liStyle} key={`${score.name}-${score.score}`}>
        <Score score={score} />
      </li>
    );
  });

  const listStyles = {
    width: '50vw'
  };
  return <ol style={listStyles}>{scoreArr}</ol>;
}

ScoreList.propTypes = {
  scores: PropTypes.array.isRequired
};

export default ScoreList;
