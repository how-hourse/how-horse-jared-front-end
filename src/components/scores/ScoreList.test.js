import React from 'react';
import { shallow } from 'enzyme';
import ScoreList from './ScoreList';

describe('ScoreList component', () => {
  it('renders a ScoreList', () => {
    const scoreArr = [
      {
        name: 'taylor',
        score: 24
      },
      {
        name: 'taylor',
        score: 24
      },
      {
        name: 'taylor',
        score: 24
      },
      {
        name: 'taylor',
        score: 24
      }
    ];


    const wrapper = shallow(<ScoreList scores={scoreArr}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
