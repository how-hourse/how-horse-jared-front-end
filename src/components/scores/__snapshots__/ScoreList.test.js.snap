// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreList component renders a ScoreList 1`] = `
<ol
  style={
    Object {
      "width": "50vw",
    }
  }
>
  <li
    key="taylor-24"
    style={
      Object {
        "padding": "0px",
      }
    }
  >
    <Score
      score={
        Object {
          "name": "taylor",
          "score": 24,
        }
      }
    />
  </li>
  <li
    key="taylor-24"
    style={
      Object {
        "padding": "0px",
      }
    }
  >
    <Score
      score={
        Object {
          "name": "taylor",
          "score": 24,
        }
      }
    />
  </li>
  <li
    key="taylor-24"
    style={
      Object {
        "padding": "0px",
      }
    }
  >
    <Score
      score={
        Object {
          "name": "taylor",
          "score": 24,
        }
      }
    />
  </li>
  <li
    key="taylor-24"
    style={
      Object {
        "padding": "0px",
      }
    }
  >
    <Score
      score={
        Object {
          "name": "taylor",
          "score": 24,
        }
      }
    />
  </li>
</ol>
`;
